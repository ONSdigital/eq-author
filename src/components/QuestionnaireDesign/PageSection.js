import React, { cloneElement, Component, Children } from "react";
import PropTypes from "prop-types";
import { noop } from "lodash";
import styled from "styled-components";
import { colors } from "constants/theme";

const StyledPageSection = styled.div`
  padding: 2em 2.625em;
  margin-bottom: -2px;
  border-bottom: 2px dashed #c6c6c6;
  outline-offset: -3px;
  outline: ${props =>
    props.focussed ? `3px solid ${colors.lightBlue}` : "none"};

  &:last-child {
    border: none;
  }
  &:hover {
    outline: 3px solid ${colors.lightBlue};
  }
`;

export default class PageSection extends Component {
  static propTypes = {
    children: PropTypes.node,
    onFocus: PropTypes.func,
    focussed: PropTypes.bool,
    id: PropTypes.string
  };

  static defaultProps = {
    onFocus: noop,
    onBlur: noop,
    focussed: false
  };

  handleFocus = e => {
    this.props.onFocus(this.props.id);
  };

  render() {
    const { children, focussed } = this.props;

    return (
      <StyledPageSection onFocus={this.handleFocus} focussed={focussed}>
        {Children.map(children, child => cloneElement(child, { focussed }))}
      </StyledPageSection>
    );
  }
}
