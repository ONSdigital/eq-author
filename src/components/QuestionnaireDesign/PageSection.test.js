import React from "react";
import { shallow } from "enzyme";
import PageSection from "./PageSection";

let component, handleFocus;

const Child = () => <div />;

describe("PageSection", () => {
  beforeEach(() => {
    handleFocus = jest.fn();

    component = shallow(
      <PageSection id="foo" onFocus={handleFocus} focussed>
        <Child />
      </PageSection>
    );
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });

  it("should handle focus event", () => {
    component.simulate("focus");
    expect(handleFocus).toHaveBeenCalledWith("foo");
  });

  it("should pass on its focussed prop", () => {
    expect(component.find(Child).prop("focussed")).toBe(true);
  });
});
