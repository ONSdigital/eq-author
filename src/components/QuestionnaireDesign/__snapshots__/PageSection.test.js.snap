// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSection should render 1`] = `
<PageSection__StyledPageSection
  focussed={true}
  onFocus={[Function]}
>
  <Child
    focussed={true}
  />
</PageSection__StyledPageSection>
`;
